### PR TITLE
rockskip: Add docs

### PR DESCRIPTION
**Pull request by chrismwendt**
Adds docs for Rockskip #28719 

## Test plan

Proofread 📝

### CODE REVIEW

**Review by mrnugget — APPROVED** [review]
Can you also link to this from the `doc/code_intelligence/index.md` file?

**Comment by chrismwendt** [comment]
✅ Added a link from `doc/code_intelligence/index.md`

File: doc/code_intelligence/explanations/rockskip.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
+# Rockskip: fast symbol sidebar and search-based code intelligence on monorepos
+
+Rockskip is an alternative symbol indexing and query engine for the symbol service intended to improve performance of the symbol sidebar and search-based code intelligence on big monorepos. It was added in Sourcegraph 3.38.
+
+## When should I use Rockskip?
+
+![still processing symbols error](https://user-images.githubusercontent.com/1387653/157811739-1d9aa6ef-b56a-4192-a4cc-41fd162efbb1.png)
```
Comment (eseliger) [inline doc/code_intelligence/explanations/rockskip.md:7]: I think this should live in our gcp bucket for assets so we control the TTL

File: doc/code_intelligence/explanations/rockskip.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+![hover popover spinner](https://user-images.githubusercontent.com/1387653/157811918-b35cab26-c07a-4ed9-ab4d-f4c0c616492b.png)
+
+If you regularly see the above error or slow hover popovers (when not using LSIF), it probably means that the default implementation (which copyies SQLite DBs for each commit) is not fast enough and that Rockskip might help.
```
Comment (macraig) [inline doc/code_intelligence/explanations/rockskip.md:11]:
```suggestion
If you regularly see the above error or slow hover popovers (when not using LSIF), it probably means that the default implementation (which copies SQLite DBs for each commit) is not fast enough and that Rockskip might help.
```

File: doc/code_intelligence/explanations/index.md [hunk 3]
```diff
@@ -4,6 +4,7 @@
 - [Precise code intelligence](precise_code_intelligence.md)
   - [Precise code intelligence uploads](uploads.md)
 - [Search-based code intelligence](search_based_code_intelligence.md)
+- <span class="badge badge-experimental">Experimental</span> [Rockskip: faster symbol searches](rockskip.md)
```
Comment (macraig) [inline doc/code_intelligence/explanations/index.md:7]: Given it's related to improving Symbol search, should it be below the feature? I'm wondering if it's clear for someone to read "Faster symbol search" before getting to the Symbol search section (just a thought, feel free to ignore)

File: doc/code_intelligence/index.md [hunk 4]
```diff
@@ -73,6 +73,7 @@ Code intelligence provides advanced code navigation features that let developers
 - [Precise code intelligence](explanations/precise_code_intelligence.md)
   - [Precise code intelligence uploads](explanations/uploads.md)
 - [Search-based code intelligence](explanations/search_based_code_intelligence.md)
+- <span class="badge badge-experimental">Experimental</span> [Rockskip: faster symbol searches](explanations/rockskip.md)
```
Comment (macraig) [inline doc/code_intelligence/index.md:76]: Same as above, still optional

**Review by macraig — APPROVED** [review]

File: doc/code_intelligence/explanations/rockskip.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+      containers:
+      - name: symbols
+        env:
+        # Enables Rockskip
+        - name: USE_ROCKSKIP
+          value: "true"
+        # Uses Rockskip for the repositories in the comma separated list
+        - name: ROCKSKIP_REPOS
+          value: "github.com/torvalds/linux,github.com/pallets/flask"
+```
```
Comment (efritz) [inline doc/code_intelligence/explanations/rockskip.md:35]: Might want a short sentence about non-k8s deployments

**Review by efritz — APPROVED** [review]
This has a lot of in-depth knowledge. Thanks!

File: doc/code_intelligence/explanations/rockskip.md (hunk 1 above)
Comment (chrismwendt) [reply doc/code_intelligence/explanations/rockskip.md:7]: Uploaded to GCP storage.

**Pull request merged**